.clang-tidy

```diff
@@ -42,259 +42,69 @@ HeaderFilterRegex: '*.(h|hpp|hxx)'
-    value:           ''
-  - key:             readability-identifier-naming.ClassPrefix
-    value:           ''
-  - key:             readability-identifier-naming.ClassSuffix
-    value:           ''
   - key:             readability-identifier-naming.ConstantCase
     value:           aNy_CasE
   - key:             readability-identifier-naming.ConstantMemberCase
     value:           aNy_CasE
-  - key:             readability-identifier-naming.ConstantMemberPrefix
-    value:           ''
-  - key:             readability-identifier-naming.ConstantMemberSuffix
-    value:           ''
   - key:             readability-identifier-naming.ConstantParameterCase
     value:           aNy_CasE
-  - key:             readability-identifier-naming.ConstantParameterPrefix
-    value:           ''
-  - key:             readability-identifier-naming.ConstantParameterSuffix
-    value:           ''
-  - key:             readability-identifier-naming.ConstantPrefix
-    value:           ''
-  - key:             readability-identifier-naming.ConstantSuffix
-    value:           ''
   - key:             readability-identifier-naming.ConstexprFunctionCase
     value:           aNy_CasE
-  - key:             readability-identifier-naming.ConstexprFunctionPrefix
-    value:           ''
-  - key:             readability-identifier-naming.ConstexprFunctionSuffix
-    value:           ''
   - key:             readability-identifier-naming.ConstexprMethodCase
     value:           aNy_CasE
-  - key:             readability-identifier-naming.ConstexprMethodPrefix
-    value:           ''
-  - key:             readability-identifier-naming.ConstexprMethodSuffix
-    value:           ''
   - key:             readability-identifier-naming.ConstexprVariableCase
     value:           aNy_CasE
-  - key:             readability-identifier-naming.ConstexprVariablePrefix
-    value:           ''
-  - key:             readability-identifier-naming.ConstexprVariableSuffix
-    value:           ''
   - key:             readability-identifier-naming.EnumCase
-    value:           lower_case
+    value:           CamelCase
   - key:             readability-identifier-naming.EnumConstantCase
-    value:           aNy_CasE
-  - key:             readability-identifier-naming.EnumConstantPrefix
-    value:           ''
-  - key:             readability-identifier-naming.EnumConstantSuffix
-    value:           ''
-  - key:             readability-identifier-naming.EnumPrefix
-    value:           ''
-  - key:             readability-identifier-naming.EnumSuffix
-    value:           ''
+    value:           CamelCase
   - key:             readability-identifier-naming.FunctionCase
-    value:           lower_case
-  - key:             readability-identifier-naming.FunctionPrefix
-    value:           ''
-  - key:             readability-identifier-naming.FunctionSuffix
-    value:           ''
+    value:           camelBack
```

Comment:
Will this break named requirements in some STL functions? e.g. `try_lock`?

Reply:
It will effect such functions, but clang-tidy will only report warnings/errors, in our code(, as long the CMakeLists.txt is configured correctly). We can write a `// NOLINT[readability-identifier-naming]` to suppress those warnings.